pallets/fragments/src/tests.rs

```diff
@@ -202,17 +202,29 @@ fn update_should_works() {
 			references,
 			None,
 			signature,
-			immutable_data,
+			immutable_data.clone(),
 		));
 
-		let signature: sp_core::ecdsa::Signature = pair.sign(&fragment_hash);
+		let data = Some(immutable_data.clone());
+		let fragment_hash = blake2_256(&immutable_data);
+		let data_hash = blake2_256(&data.encode());
+		let signature_hash = blake2_256(&[&fragment_hash[..], &data_hash[..]].concat());
+
+		let signature: sp_core::ecdsa::Signature = pair.sign(&signature_hash);
+
+		let recover = Crypto::secp256k1_ecdsa_recover_compressed(&signature.0, &signature_hash)
+			.ok()
+			.unwrap();
+		let recover = sp_core::ecdsa::Public(recover);
+
+		assert_ok!(FragmentsPallet::add_upload_auth(Origin::root(), recover, who));
```

Comment:
This doesn't test anything and it's wrong, just makes test pass, we are adding the recovered key into auth, it's wrong, we should just add `//Charlie` to auth. Nothing else to test proper signature workflow.

Reply:
Please fix in the common code we talked about. 🙏

Reply:
@sinkingsugar We need this to satisfy this condition https://github.com/fragcolor-xyz/clamor/blob/master/pallets/fragments/src/lib.rs#L301

Reply:
@ashi20192020ayush correct but the signer is an external service off chain not the users.
So it needs to be just 1 key e.g. `//Charlie`

Reply:
Users will ask to that service to verify the payload they upload is valid and receive a signature if valid, to allow them to transact basically.
Sorry this wasn't too clear.